src/briefcase/integrations/android_sdk.py

```diff
@@ -0,0 +1,370 @@
+import re
+import shutil
+import subprocess
+from pathlib import Path
+
+from requests import exceptions as requests_exceptions
+
+from briefcase.exceptions import BriefcaseCommandError, NetworkFailure
+
+DEVICE_NOT_FOUND = re.compile(r"^error: device '[^']*' not found")
+
+
+def verify_android_sdk(command):
+    """
+    Verify an Android SDK is available.
+
+    If the ANDROID_SDK_ROOT environment variable is set, that location will
+    be checked for a valid SDK.
+
+    If the location provided doesn't contain an SDK, or no location is provided,
+    an SDK is downloaded.
+
+    :param command: The command making the verification request.
+    :returns: An AndroidSDK instance, bound to command.
+    """
+    sdk_root = command.os.environ.get('ANDROID_SDK_ROOT')
+    if sdk_root:
+        sdk = AndroidSDK(command=command, root_path=Path(sdk_root))
+
+        if sdk.exists():
+            # Ensure licenses have been accepted
+            sdk.verify_license()
+            return sdk
+        else:
+            print("""
```

Comment:
I'm -0 on printing something out every time, but it's OK. I get why you're doing it. I don't know that I really have a better idea, but it seems a little sadly noisy.

Reply:
I agree that being noisy isn't generally a good thing; but in this case, it will only be noisy if (a) you have `ANDROID_SDK_ROOT` set and (b) it's not actually pointing at an Android SDK. If that's happening, I don't feel especially bad about yelling at the user to fix their setup.